core/crypto/value_test.go

```diff
@@ -0,0 +1,46 @@
+package crypto
+
+import "testing"
+
+func TestGenerateValue(t *testing.T) {
+	type args struct {
+		template string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Success case for alphanumeric",
+			args: args{
+				template: `foo[\w]{8}bar`,
+			},
+			wantErr: false,
+		},
+		{
+			name: "Fail case for invalid regex",
+			args: args{
+				template: `pass[z-a]{8}`,
+			},
+			wantErr: true,
```

Comment:
Also, it would be good if we can match exact returned error. Since we've multiple err scenario in this function.

Reply:
I note it I can fix that